heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesScheduler.java

```diff
@@ -0,0 +1,247 @@
+    ObjectNode metadataNode = mapper.createObjectNode();
+    metadataNode.put(KubernetesConstants.NAME, Joiner.on("-").join(Runtime.topologyName(runtime), containerIndex));
+
+    ObjectNode labels = mapper.createObjectNode();
+    labels.put(KubernetesConstants.TOPOLOGY_LABEL, Runtime.topologyName(runtime));
+
+    metadataNode.set(KubernetesConstants.METADATA_LABELS, labels);
+
+    return metadataNode;
+  }
+
+  // build the metadata for a pod template
+  protected ObjectNode getPodMetadata(ObjectMapper mapper, int containerIndex) {
+    ObjectNode podMetadata = mapper.createObjectNode();
+    ObjectNode labelData = mapper.createObjectNode();
+
+    labelData.put(KubernetesConstants.APP, Integer.toString(containerIndex));
+    podMetadata.set(KubernetesConstants.METADATA_LABELS, labelData);
+
+    return podMetadata;
+  }
+
+  // build the container spec for the deployment
+  protected ObjectNode getContainerSpec(ObjectMapper mapper,
+                                        int containerIndex,
+                                        Resource containerResource) {
+
+    ObjectNode containerSpec = mapper.createObjectNode();
+    ArrayNode containerList = mapper.createArrayNode();
+
+    ObjectNode containerInfo = mapper.createObjectNode();
+    containerInfo.put(KubernetesConstants.NAME, Joiner.on("-").join("executor", Integer.toString(containerIndex)));
+
+    // set the host for this container
+    ArrayNode envList = mapper.createArrayNode();
+    ObjectNode envVar = mapper.createObjectNode();
+    envVar.put(KubernetesConstants.NAME, KubernetesConstants.HOST);
+
+    // build the JSON to attach the Pod IP as the "HOST" environment variable
+    ObjectNode fieldRef = mapper.createObjectNode();
+    ObjectNode fieldPath = mapper.createObjectNode();
+    fieldPath.put(KubernetesConstants.FIELD_PATH, KubernetesConstants.POD_IP);
+    fieldRef.set(KubernetesConstants.FIELD_REF, fieldPath);
+    envVar.set(KubernetesConstants.VALUE_FROM, fieldRef);
+    envList.add(envVar);
+
+    containerInfo.set(KubernetesConstants.ENV, envList);
+
+    // Image information for this container
+    containerInfo.put(KubernetesConstants.DOCKER_IMAGE, KubernetesContext.getExecutorDockerImage(config));
+
+    // Port information for this container
+    containerInfo.set(KubernetesConstants.PORTS, getPorts(mapper));
+
+    // Heron command for the container
+    String[] command = getExecutorCommand(containerIndex);
+    ArrayNode commandsArray = mapper.createArrayNode();
+    for (int i = 0; i < command.length; i++) {
+      commandsArray.add(command[i]);
+    }
+    containerInfo.set(KubernetesConstants.COMMAND, commandsArray);
+
+    // Requested resource info
+    ObjectNode requestedResourceInfo = mapper.createObjectNode();
+    requestedResourceInfo.put(KubernetesConstants.MEMORY, containerResource.getRam().asMegabytes());
+    requestedResourceInfo.put(KubernetesConstants.CPU, containerResource.getCpu());
```

Comment:
got it. We can add this once the feature is available -
https://github.com/twitter/heron/issues/1943